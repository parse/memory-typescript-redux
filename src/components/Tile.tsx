import * as React from 'react';
import { Tile as TileType } from '../types/index';

interface TileProps {
  index: number;
  onClickTile: (tile: TileType, index: number) => void;
  tile: TileType;
}

class Tile extends React.Component<TileProps, {}> {
  render() {
    const { tile, index, onClickTile } = this.props;
    let classFlipped = tile.flipped
      ? 'tile effect__click flipped'
      : 'tile effect__click';

    const tileBackgroundImage = {
      backgroundImage: 'url(' + tile.image + ')',
    };

    return (
      <div
        className={classFlipped}
        onClick={() => {
          onClickTile(tile, index);
        }}
      >
        <div className="tile__front" />
        <div className="tile__back" style={tileBackgroundImage} />
      </div>
    );
  }
}

export default Tile;
