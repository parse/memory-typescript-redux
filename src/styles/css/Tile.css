.tile {
  position: relative;
  float: left;
  padding-bottom: 25%;
  width: 25%;
  text-align: center; }
  .tile__front {
    cursor: pointer;
    background-image: url("/assets/img/deck.jpg");
    background-size: cover; }
  .tile__back {
    -webkit-transform: rotateY(-180deg);
    transform: rotateY(-180deg);
    background-size: 90% 90%;
    background-position: center;
    background-repeat: no-repeat; }
  .tile__front, .tile__back {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: 10px;
    border: 1px solid #888;
    -webkit-backface-visibility: hidden;
    backface-visibility: hidden;
    -webkit-transition: -webkit-transform 0.3s;
    transition: transform 0.3s; }
  .tile.effect__click.flipped .tile__front {
    -webkit-transform: rotateY(-180deg);
    transform: rotateY(-180deg); }
  .tile.effect__click.flipped .tile__back {
    -webkit-transform: rotateY(0);
    transform: rotateY(0); }

/*# sourceMappingURL=Tile.css.map */
