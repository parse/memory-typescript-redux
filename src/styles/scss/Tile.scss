.tile {
  position: relative;
  float: left;
  padding-bottom: 25%;
  width: 25%;
  text-align: center;

  &__front {
    cursor: pointer;
    background-image: url('/assets/img/deck.jpg');
    background-size: cover;
  }

  &__back {
    -webkit-transform: rotateY(-180deg);
            transform: rotateY(-180deg);
            background-size: 90% 90%;
            background-position: center;
            background-repeat: no-repeat;
  }

  &__front,
  &__back {
     position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: 10px;
    border: 1px solid #888;

    -webkit-backface-visibility: hidden;
          backface-visibility: hidden;
    -webkit-transition: -webkit-transform 0.3s;
          transition: transform 0.3s;
  }

  &.effect__click.flipped {

    & .tile__front {
      -webkit-transform: rotateY(-180deg);
      transform: rotateY(-180deg);
    }

    & .tile__back {
      -webkit-transform: rotateY(0);
      transform: rotateY(0);
    }

  }

}
