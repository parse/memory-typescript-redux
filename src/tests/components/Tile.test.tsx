import * as React from 'react';
import Tile from '../../components/Tile';
import * as renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tile = {
    image: 'lol.png',
    tileKey: 1,
    flipped: false,
    matched: false,
  };

  const tree = renderer
    .create(<Tile tile={tile} index={1} onClickTile={() => null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
