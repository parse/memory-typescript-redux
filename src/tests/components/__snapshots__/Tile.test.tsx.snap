// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tile effect__click"
  onClick={[Function]}
>
  <div
    className="tile__front"
  />
  <div
    className="tile__back"
    style={
      Object {
        "backgroundImage": "url(lol.png)",
      }
    }
  />
</div>
`;
